src/sql/workbench/contrib/query/common/queryResultsWriterStatus.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+
+export enum QueryResultsWriterMode {
+	ToGrid,
+	ToFile,
+}
+
+export class QueryResultsWriterStatus extends Disposable {
+	public static instance: QueryResultsWriterStatus = undefined;
```

Comment:
You should avoid this kind of singleton pattern. If a thing is a singleton then it should be a service instead.

But in general this doesn't seem like it should be a singleton anyways. And I'm not even sure why it's necessary - each grid should keep track of where it's currently outputting that are currently enabled for it. This shouldn't be a global thing for all editors.

Reply:
When I diffed everything from the other PR it must have picked this up. I've removed it since I'm not using a singleton to track output status modes for different editors. Those are now being tracked on a per editor basis.